Vector.c

```diff
@@ -62,14 +62,6 @@ static bool Vector_isConsistent(const Vector* this) {
    assert(this->items <= this->arraySize);
    assert(!Vector_isDirty(this));
 
-   if (this->owner) {
-      for (int i = 0; i < this->items; i++) {
-         if (!this->array[i]) {
-            return false;
-         }
-      }
-   }
-
```

Comment:
We could count the non empty slots:
```
int items = 0;
for (int i = 0; i < this->arraySize; i++)
   if (this->array[i])
      items++;
assert(this->items == items);
```

This requires `memset(this->array, '\0', this->arraySize * sizeof(Object *));` in `Vector_prune()`

Reply:
I don't think that this will work.

`this->items` does currently just contain the actual length of the vector and not the number of used slots.

Look for example at `Vector_set()`:
`Vector_set()` allows calls with `idx` >`this->items`. In this case `this->items` is just set to `idx +1`. 
`this->items` would then also include empty slots and the assertion would fail.

Reply:
@cgzones any thoughts on this?

Reply:
@cgzones There are open questions that you might want to take a look at.

Reply:
> I don't think that this will work.
>
>this->items does currently just contain the actual length (or in other words the first unused index) of the array and >not the number of used slots.
>
>Look for example at Vector_set():
>Vector_set() allows calls with idx >this->items. In this case this->items is just set to idx +1.
>this->items would then also include empty slots and the assertion would fail.

True, so disregard my suggestion.

> The memset in Vector_prune() might be a good idea anyway.

Could you add this?

Reply:
done